modules/apps/healthcheck/healthcheck-api/src/main/java/com/liferay/healthcheck/HealthcheckItem.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.healthcheck;
+
+/**
+ * @author Olaf Kock
+ */
+public class HealthcheckItem {
+
+	public HealthcheckItem(
+		boolean resolved, String sourceKey, String link, String messageKey,
+		Object... messageParameters) {
+
+		_resolved = resolved;
+		_sourceKey = sourceKey;
+		_link = link;
+		_messageKey = messageKey;
+		_messageParameters = messageParameters;
+	}
+
+	public HealthcheckItem(Object cause, Throwable throwable) {
+		String className = cause.getClass(
+		).getName();
+
+		_resolved = false;
+		_sourceKey = className + "-exception";
+		_link = null;
+		_messageKey = "an-exception-occurred-for-x-x-x";
+		_messageParameters = new Object[] {
+			className,
+			throwable.getClass(
+			).getName(),
+			throwable.getMessage()
+		};
+	}
+
+	/**
+	 * A link(URL) that can contain further information on the tested condition
+	 *
+	 * @return a link URL
+	 */
+	public String getLink() {
+		return _link;
+	}
+
+	/**
+	 * A message key that describes what this healthcheck checks for
+	 *
+	 * @return human readable message
+	 */
+	public String getMessageKey() {
+		return _messageKey;
+	}
+
+	/**
+	 * Parameters that are used to look up the human readable message,
+	 * with the messageKey.
+	 * @return an array of parameters to be combined with the localized message
+	 */
+	public Object[] getMessageParameters() {
+		return _messageParameters;
+	}
+
+	/**
+	 * A machine readable key that can be used to refer to a particular healthcheck
+	 * or its result. This was introduced to be able to ignore certain healthchecks,
+	 * in case their test does not apply to a certain environment (example:
+	 * Elasticsearch Sidecar is ok in local demo systems). Default content: The
+	 * healthcheck's fully qualified classname, optionally extended by extra
+	 * information (each healthcheck might execute several checks) and the
+	 *
+	 * @return the machine readable encoding for this healthcheck
+	 */
+	public String getSourceKey() {
+		return _sourceKey + "-" + _resolved;
```

Comment:
@olafk this getSourceKey() is confusing. As far as I can see, it is only used for the ignore purpose. However, if same HealthCheck can generate multiple HealthCheckItem with same resolved state, this getSourceKey() will return the same result. For example, AvailableLocaleConfigurationHealthcheck can do this. When that happens, if you ignore one item of that source key, all items from AvailableLocaleConfigurationHealthcheck will be ignored.

I am not sure whether it is expected.
If yes, the ignore should be handled at HealthCheck level, rather than HealthCheckItem level.
If no, the key must be able to uniquely identify one HealthCheckItem.

I think it makes more sense to be "yes" case, otherwise properties change might lead to unreachable HealthCheckItem unique key, causing leftover data.

Reply:
Those are indeed meant to be for a single healthcheck to return multiple results that can be ignored one-by-one. The trivial checks that I've included in this PR just didn't contain an example for this use case.
I've changed AvailableLocalesConfigurationHealthcheck to implement this case, so that it makes sense to have this level of complexity already in. It  will make a lot more sense when it comes to a number of individual host names for client extensions and data providers (a later addition that is already implemented, but not yet pullrequested).

I have pushed an updated branch (with --force), not sure if this change is reflected in this PR.

Reply:
right - this still comes from a previous exploratory implementation where I've introduced it to assist with a quick test. Eliminating. 
With that, I'm eliminating all of the external construction of sourceKey, and just combine the message-key and the parameters. Will push the changes tomorrow. Thank you for triggering this insight, this simplifies _all_ implementations further.